Auto-apply column filters from popover

TomTirapani opened the pull request:
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

lbwexler commented on cmp/grid/filter/GridFilterModel.js:
```diff
@@ -22,6 +22,8 @@ export class GridFilterModel extends HoistModel {
     /** @member {(Store|View)} */
     bind;
+    /** @member {boolean} */
+    autoApply;
```
should this be `immediateApply` it does still get `auto`matically applied to the bound store

lbwexler replied:
or perhaps, `commitOnChange` ?

TomTirapani replied:
Will change to `commitOnChange`... I notice that's what we use for the equivalent setting on the Column Chooser

lbwexler commented on desktop/cmp/grid/impl/filter/ColumnHeaderFilterModel.js:
```diff
@@ -38,15 +38,20 @@ export class ColumnHeaderFilterModel extends HoistModel {
 
+    @computed.struct
+    get pendingFilter() {
+        const {activeTabId} = this.tabContainerModel;
```
does this yield an immediate filter change just from changing tabs?   That seems surprising.

TomTirapani replied:
Good point! :|

Not sure what the best approach is here. I could tweak the logic to wait until you've **made a change** in the new tab to commit the filter?

Alternatively, this could be a good justification to make require the user to explicitly clear out their filter upon switching tabs, which we currently only do when switching to the Values Tab from the Custom Tab.

lbwexler replied:
I think we could just  flip things around so that each UI tab puts on its own reaction, such that *when it is visible*, and its value changes, then it should do the commit action

lbwexler replied:
So playing with this.  Its really nice when the data set is the right size, and its all snappy.  

One issue I have is that for the custom tab, its now updating on every keystroke.  This to me seems overly aggressive.  Its especially problematic when you are trying to type in, say, a number, and you get intermediate results based on the first digits that you type.

Think we could potentially make the text fields on the Custom tab commit on change: false, so you had to hit enter, or leave the field for the new value to flow through.  (f we did that, would also probably need to always enable the apply button as well).

I also see a flash in the FilterChooser (in our toolbox example tab) whenever I update the filter via the single keystrokes.  Not sure what that is, but its annoying.  We might address the underlying cause, but that will be less important if we change the behavior as described above.

lbwexler replied:
Could be good to add a little switch at the bottom of the toolbox example just to turn this `commitOnChange` behavior on and off

TomTirapani commented:
Thanks for continuing to look at this Lee!

I've adjusted the commitOnChange behaviour of the inputs on the custom tab, and added the toggle to Toolbox example to help with testing.

Thanks also for spotting that FilterChooser issue. That was not directly related to this change, just more apparent due to the commitOnChange. I've opened as separate PR which fixes that: https://github.com/xh/hoist-react/pull/2638
